src/Ar/StringExt/stringdtoa.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * File: time.cpp
+ * Copyright (c) 2023 Loupe
+ * https://loupe.team
+ * 
+ * This file is part of StringExt, licensed under the MIT License.
+ * 
+ */
+
+#include <bur/plctypes.h>
+#ifdef __cplusplus
+	extern "C"
+	{
+#endif
+#include "StringExt.h"
+#ifdef __cplusplus
+	};
+#endif
+
+#include "string.h"
+
+#include "./gdtoa/gdtoa.h"
+
+
+unsigned long stringdtoa(double value, plcstring* buffer, unsigned long ndig, unsigned long bufferSize) {	
+	return g_dfmt(buffer, &value, ndig, bufferSize);
+}
+unsigned long stringftoa(float value, plcstring* buffer, unsigned long ndig, unsigned long bufferSize) {	
+	return g_ffmt(buffer, &value, ndig, bufferSize);
+}
+
+double stringatod(plcstring* value, plcstring** pEnd) {
+	return strtod(value, pEnd);
+}
+float stringatof(plcstring* value, plcstring** pEnd) {
+	return strtof(value, pEnd);
+}
```

Comment:
I'm not sure I understand how pEnd is supposed to be used. Naively, It feels like there may be a friendlier api we could implement in this layer. Perhaps a string length or size being passed into stringatof/d?

Reply:
Fair point. pEnd will be populated with a pointer to the end of the number. So its an output. We could instead return a length something like:
```c
float stringatof(plcstring* value, UDINT* pLen) {
	char* end = 0;
	float val = strtof(value, end);
	if(pEnd) *pEnd = (value-end);
	return val;
}
````

Reply:
From linux man page:
> double atof(const char *nptr);
> 
> Description
> 
> The atof() function converts the initial portion of the string pointed to by nptr to double. The behavior is the same as
> 
>     strtod(nptr, NULL); 
> except that atof() does not detect errors.

It seems this implementation is closer to strtof/d than atof/d.
I think we could either keep the pEnd argument and update the name to indicate we are implementing strtof/d, or we could throw out the error checking and have the wrapper take only the value argument like standard atof/d functions.

In fact, I might be in favor of implementing both here - stringatof/d and stringstrtof/d
One provides error feedback via pEnd and the other is a slicker api

Reply:
I renamed them to stringstrtof/d. stringatof/d can be added in the future